include/natalie/match_data_object.hpp

```diff
@@ -33,10 +33,11 @@ class MatchDataObject : public Object {
 
     size_t size() { return m_region->num_regs; }
 
-    size_t index(size_t);
+    long index(size_t);
```

Comment:
This is important because the index can be `-1` if the capture group doesn't have a value, and it was underflowing with `size_t` previously.

Reply:
We usually use `ssize_t` if it can be negative.

Reply:
Updated!